src/formatDate.js

```diff
@@ -50,7 +50,44 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+  const oldFormat = fromFormat.splice(0, 3);
+  const newFormat = toFormat.splice(0, 3);
+  const rawDate = date.split(oldSeparator);
+
+  let newDate = '';
+
+  for (let j = 0; j < newFormat.length; j++) {
+    for (let i = 0; i < oldFormat.length; i++) {
+      if (newFormat[j] === oldFormat[i]) {
+        newDate += rawDate[i];
+        break;
+      }
+
+      if (i === oldFormat.length - 1
+        && newFormat[j] !== oldFormat[i]
```

Comment:
I think, given that the condition is repeated, it is better to make a separate if

Reply:
I am not sure if I got you right, but I managed to shorten the code :) Thanks!